src/components/attribute/EmptyState.tsx

```diff
@@ -0,0 +1,174 @@
+import { IDENTIFIER } from "@lib/extension";
+import React, { useEffect, useState } from "react";
+
+import { useClipboard } from "@lib/useClipboard";
+
+import { updatePullRequestLinkOnRecord } from "@lib/fields";
+import { withGitHubApi } from "@lib/github/api";
+import { getPrByUrl } from "@lib/github/getPr";
+import { LinkableRecord } from "@lib/linkableRecord";
+import { validPrUrl } from "@lib/validPrUrl";
+
+const LEARN_MORE_URL = "https://github.com/aha-develop/github";
+const ICON = aha.iconForExtensionIdentifier(IDENTIFIER);
+
+type MenuProps = {
+  record: LinkableRecord;
+  onPaste: Function;
+};
+
+const Menu = ({ record, onPaste }: MenuProps) => {
+  const handleSync = () => {
+    aha.command(`${IDENTIFIER}.sync`, { record });
+  };
+
+  return (
+    <aha-menu>
+      <aha-button slot="button" type="attribute" size="mini">
+        <aha-icon icon="fa-solid fa-ellipsis"></aha-icon>
+      </aha-button>
+      <aha-menu-item onClick={() => onPaste()}>Paste PR link</aha-menu-item>
+      <aha-menu-item onClick={handleSync}>Scan GitHub</aha-menu-item>
+      <hr />
+      <aha-menu-item>
+        <a href={LEARN_MORE_URL} target="_blank" rel="noopener noreferrer">
+          <aha-icon icon="fa fa-external-link" />
+          Read the docs
+        </a>
+      </aha-menu-item>
+    </aha-menu>
+  );
+};
+
+const EmptyStateBox = ({ children }) => (
+  <aha-box class="m-0" style={{ color: "var(--theme-secondary-text)" }}>
+    <div style={{ margin: "calc(-2em + 12px)" }}>{children}</div>
+  </aha-box>
+);
+
+export const EmptyState: React.FC<{ record: LinkableRecord }> = ({
+  record,
+}) => {
+  const [onCopy, copied] = useClipboard();
+  const [pasteMode, setPasteMode] = useState<Boolean>(false);
+  const [validation, setValidation] = useState<String | null>(null);
+  const [hasConfiguredWebhook, setHasConfiguredWebhook] =
+    useState<Boolean | null>(null);
+
+  // aha-menu gets mad if you remove it from the DOM while the menu is open.
+  // Wait 1 tick so the menu has a chance to unmount first.
+  const viewPasteMode = () => {
+    setTimeout(() => setPasteMode(true), 1);
+  };
+
+  const pasteLink = async (url: string) => {
+    if (!validPrUrl(url)) {
```

Comment:
Same as on the GitLab extension - `new URL()` throws if I put a junk string in, so the user doesn't get any validation feedback. `try/catch`?

Reply:
done